File: 1179.c
#include <stdio.h>

int main() {
    int par[5], impar[5], x, cp = 0, ci = 0;
    
    for (int i = 0; i < 15; i++) {
        scanf("%d", &x);
        if(x % 2 != 0) {
            // x é ímpar
            if (ci < 5) impar[ci] = x;
            if (ci++ == 4) {
                int c = 0;
                while (c < 5) {
                    printf("impar[%d] = %d\n", c, impar[c]);
                    impar[c++] = 0;
                }
                ci = 0;
            }
        } else {
            // x é par
            if (cp < 5) par[cp] = x;
            if (cp++ == 4) {
                int c = 0;
                while (c < 5) {
                    printf("par[%d] = %d\n", c, par[c]);
                    par[c++] = 0;
                }
                cp = 0;
            }
        }
    }
    
    for (int i = 0; i < ci; i++)
        if (impar[i] != 0) printf("impar[%d] = %d\n", i, impar[i]);
    
    for (int i = 0; i < cp; i++)
        if (par[i] != 0) printf("par[%d] = %d\n", i, par[i]);
    
    return 0;
}

File: 1179.py
"""
POR TUDO O QUE É MAIS SAGRADO, NÃO ENCURTE O CÓDIGO, COMO: cp = ci = 0, POIS ISSO DÁ ERRO!
"""

par = [0, 0, 0, 0, 0]
impar = [0, 0, 0, 0, 0]
cp = 0
ci = 0

for i in range(0, 15):
    x = int(input(""))
    
    if (x % 2 == 0):
        par[cp] = x
        cp = cp + 1
        if (cp == 5):
            for j in range(0, 5):
                print(f"par[{j}] = {par[j]}")
                par[j] = 0
            cp = 0
    else:
        impar[ci] = x
        ci += 1
        if (ci == 5):
            for j in range(0, 5):
                print(f"impar[{j}] = {impar[j]}")
                impar[j] = 0
            ci = 0
    
for i in range(ci):
    print(f"impar[{i}] = {impar[i]}")
    
for i in range(cp):
    print(f"par[{i}] = {par[i]}")
